docs/reference/codebase-structure.md

```diff
@@ -0,0 +1,147 @@
+# Codebase structure
+
+Let's examine the Feast codebase.
+
+```
+$ tree -L 1 -d
+.
+├── docs
+├── examples
+├── go
+├── infra
+├── java
+├── protos
+├── sdk
+└── ui
+```
+
+## Python SDK
+
+The Python SDK lives in `sdk/python/feast`.
+
+```
+$ cd sdk/python/feast
+$ tree -L 1 
+.
+├── __init__.py
+├── aggregation.py
+├── base_feature_view.py
+├── batch_feature_view.py
+├── cli.py
+├── constants.py
+├── data_format.py
+├── data_source.py
+├── diff
+├── dqm
+├── driver_test_data.py
+├── embedded_go
+├── entity.py
+├── errors.py
+├── feast_object.py
+├── feature.py
+├── feature_logging.py
+├── feature_server.py
+├── feature_service.py
+├── feature_store.py
+├── feature_view.py
+├── feature_view_projection.py
+├── field.py
+├── flags_helper.py
+├── importer.py
+├── inference.py
+├── infra
+├── loaders
+├── names.py
+├── on_demand_feature_view.py
+├── online_response.py
+├── project_metadata.py
+├── proto_json.py
+├── py.typed
+├── registry.py
+├── registry_store.py
+├── repo_config.py
+├── repo_contents.py
+├── repo_operations.py
+├── repo_upgrade.py
+├── request_feature_view.py
+├── saved_dataset.py
+├── stream_feature_view.py
+├── templates
+├── transformation_server.py
+├── type_map.py
+├── types.py
+├── ui
+├── ui_server.py
+├── usage.py
+├── utils.py
+├── value_type.py
+├── version.py
+└── wait.py
+```
+
+The majority of Feast logic lives in these Python files:
+* The core Feast objects ([entities](../getting-started/concepts/entity.md), [feature views](../getting-started/concepts/feature-view.md), [data sources](../getting-started/concepts/dataset.md), etc.) are defined in their respective Python files, such as `entity.py`, `feature_view.py`, and `data_source.py`.
+* The feature store object is defined in `feature_store.py` and the associated configuration objects are defined in `repo_config.py`.
```

Comment:
map this to what users know about? aka this is what `feature_store.yaml` configuration maps to in memory